### PR TITLE
feat(testing/fixtures): Add logs to state fixtures

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description

Adds the transaction receipt to the state fixture format for it to contain the logs emitted during the execution of the transaction and make debugging easier.

For now, only the state tests contain this information, but blockchain tests do no due to the potential amount of extra information which might be not as useful in a multi-tx context.

This PR also cherry-picks https://github.com/ethereum/execution-specs/commit/42dcb0f5215b5d71322a53852bd54ed83488caaa to see the actual logs in the state test fixtures. 

Receipt example:
```json
"receipt": {
    "transactionHash": "0xdb1e409d11d92e6e8b3825ec82dff14f3661f1247c0d306ed4ff6aa22b0987f4",
    "gasUsed": "0xab63",
    "logsBloom": "0x04000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008000000000000000000000",
    "logs": [
        {
            "address": "0x1000000000000000000000000000000000000000",
            "data": "0x00",
            "topics": [
                "0x0000000000000000000000000000000000000000000000000000000000000002"
            ]
        }
    ]
},
```

## 🔗 Related Issues or PRs

N/A.

## ✅ Checklist

- [ ] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx tox -e static
    ```
- [ ] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-specs/blob/HEAD/docs/) directory.
- [ ] All: Set appropriate labels for the changes (only maintainers can apply labels).
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.
- [ ] Tests: For PRs implementing a missed test case, update the [post-mortem document](/ethereum/execution-specs/blob/HEAD/docs/writing_tests/post_mortems.md) to add an entry the list.
- [ ] Ported Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) or [tests/static](/ethereum/execution-specs/blob/HEAD/tests/static) have been assigned `@ported_from` marker.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by fselmo — COMMENTED** [review]
@marioevz looks great. We just need static checks to pass / fixed here. I was also thinking it would be good to get a test case here where the logs are not empty. It seems we only unit test empty logs.

**Comment by jochem-brouwer** [comment]
The JSON receipt here does not include the "status" field (either 0 or 1, depending on if the outer call frame did rollback or not (revert/OOG/invalid))
The `gasUsed` field is the "cumulative gas used" field in the receipt. (so for state tests this will indeed match the gas used). To avoid confusion I'd suggest using `cumulativeGasUsed` to make clear that this is the receipt field meant.

**Comment by jochem-brouwer** [comment]
I think it is also helpful to include the receipt as RLP besides the JSON one :thinking: 
The JSON might look correct on a client, but once it is serialized and then hashed it might still lead to a different receipt root (for instance extra zero bytes or not).
Not sure if helpful or if it is hard to add. @jwasinger ?

**Comment by jochem-brouwer** [comment]
Oh right, one more question, what if you produce state fixtures for pre-Byzantium forks? I suggested adding `status` field but this was not available pre-Byzantium, it was a post-tx state root. So if that's problematic then it's also not super important to add the status field of the receipt.

**Comment by jochem-brouwer** [comment]
One final comment (final last words lol): EIP-4844 has extra fields in the receipt, see:

https://eips.ethereum.org/EIPS/eip-4844#blob-transaction

> The [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) `ReceiptPayload` for this transaction is `rlp([status, cumulative_transaction_gas_used, logs_bloom, logs])`.

File: packages/testing/src/execution_testing/test_types/receipt_types.py [hunk 1]
```diff
@@ -44,25 +44,25 @@ class TransactionReceipt(CamelModel):
...
-            data.pop("post_state", None)
-            data.pop("postState", None)
             # geth (1.16+) returns extra fields in receipts
             data.pop("type", None)
             data.pop("blockNumber", None)
         return data
 
     transaction_hash: Hash | None = None
-    gas_used: HexNumber | None = None
+    post_state: Hash | None = Field(
```
Comment (fselmo) [inline packages/testing/src/execution_testing/test_types/receipt_types.py:53]: I like these changes 👌🏼

**Review by fselmo — APPROVED** [review]
lgtm, thanks! 👍🏼
